adapters/clients/replication.go

```diff
@@ -43,37 +45,87 @@ func NewReplicationClient(httpClient *http.Client) replica.Client {
 }
 
 // FetchObject fetches one object it exits
-func (c *replicationClient) FetchObject(ctx context.Context, hostName, indexName,
-	shardName string, id strfmt.UUID, selectProps search.SelectProperties,
+func (c *replicationClient) FetchObject(ctx context.Context, host, index,
+	shard string, id strfmt.UUID, selectProps search.SelectProperties,
 	additional additional.Properties,
 ) (objects.Replica, error) {
-	return objects.Replica{}, fmt.Errorf("not implemented")
+	resp := objects.Replica{}
+	req, err := newHttpReplicaRequest(ctx, http.MethodGet, host, index, shard, "", id.String(), nil)
+	if err != nil {
+		return resp, fmt.Errorf("create http request: %w", err)
+	}
+	err = c.doCustomMarshal(c.timeoutUnit*90, req, nil, &resp)
+	return resp, err
 }
 
```

Comment:
It would be nice if we had some unit tests to demonstrate that the endpoints are working as expected.

Reply:
added